### PR TITLE
rpc: add a non binary get_transaction_pool_hashes RPC

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by ph4r05** [comment]
Good catch! Sorry, I didn't realize the header is duplicated here.

File: src/rpc/core_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -1509,11 +1509,39 @@ namespace cryptonote
...
   {
     uint64_t blob_size;
     uint64_t fee;
     uint64_t time_in_pool;
+
+    BEGIN_KV_SERIALIZE_MAP()
+      KV_SERIALIZE(blob_size)
+      KV_SERIALIZE(fee)
+      KV_SERIALIZE(time_in_pool)
+    END_KV_SERIALIZE_MAP()
```
Comment (stoffu) [inline src/rpc/core_rpc_server_commands_defs.h]: Since this fix doesn't seem to be related to the main point addressed here, shouldn't it be made as a separate commit for clarity? Also, does this mean that this daemon RPC wasn't functioning properly previously?

Comment (moneromooo-monero) [reply src/rpc/core_rpc_server_commands_defs.h]: Unrelated, I'd started making a non binary version of that one too and decided otherwise since it's less useful. It's working fine AFAIK.
